compiler/src/accesses/array_access.rs

```diff
@@ -43,6 +43,10 @@ impl<'a> Program<'a> {
     ) -> Result<Value> {
         let full_length = match array.get_type() {
             Some(Type::Array(_, length)) => length,
+            // TODO: change this to either resolving an array reference or
```

Comment:
@gluax @0rphon Honestly I don't know what to do about this place. We briefly discussed it on a Leo sync, and currently our type system cannot resolve this on AST/ASG level, so the solution here is to throw an error or add additional IR instruction to define array length by ref which is definitely the wrong way.

Reply:
An error seems like a decent conservative choice.